src/iidm/converter/xml/NetworkXml.cpp

```diff
@@ -267,9 +268,12 @@ std::unique_ptr<Anonymizer> NetworkXml::write(std::ostream& ostream, const Netwo
 
     powsybl::xml::XmlStreamWriter writer(ostream, options.isIndent());
 
-    // FIXME(sebalaig): for now on, only one kind of anonymizer = FakeAnonymizer
-    // later, a real anonymizer will be instantiated depending on options.isAnonymized()
-    std::unique_ptr<Anonymizer> anonymizer(stdcxx::make_unique<FakeAnonymizer>());
+    std::unique_ptr<Anonymizer> anonymizer;
+    if (options.isAnonymized()) {
+        anonymizer = stdcxx::make_unique<SimpleAnonymizer>();
+    } else {
+        anonymizer = stdcxx::make_unique<FakeAnonymizer>();
+    }
```

Comment:
Use a ternary operator? If you have an issue with the return type, use `make_unique<Anonymizer, SimpleAnonymizer>` and `make_unique<Anonymizer, FakeAnonymizer>`